pycipapi/rest_client.py

```diff
@@ -28,13 +28,16 @@ class BlockedCase(HTTPError):
 
 def requests_retry_session(retries=5, backoff_factor=0.8, status_forcelist=(500, 502, 504, 503), session=None):
     session = session or requests.Session()
-    retry = Retry(
-        total=retries,
-        read=retries,
-        connect=retries,
-        backoff_factor=backoff_factor,
-        status_forcelist=status_forcelist,
-    )
+    if sys.version_info[0] < 3:
+        retry = retries
```

Comment:
@Staberinde, are you having problems with `from urllib3.util.retry import Retry`? we have used extensively with python2 without any problem. The solution proposed does not work in the way intended as doesn't take in account, status_forcelist and backoof_factor

Reply:
I've been seeing the following error on retry until I made this change: 
```
2019-11-11 17:03:47,882 - ERROR - componentrunner::_run_component run_id:5f3290ff05f329b3102ba891792c2307 Exception in component! component: [dispatch_transfer], is_multisample: yes, message: unsupported operand type(s) for -=: 'Retry' and 'int'
Traceback (most recent call last):
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/lib/python2.7/site-packages/bertha/compute/componentrunner.py", line 129, in _run_component
    (profiles, additional_files_to_register) = module.Component().run(params)
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/bertha/compute/workflow/component/basecomponent.py", line 41, in run
    self.main(params)
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/bertha/compute/workflow/component/dispatch_transfer/component.py", line 283, in main
    self.profiles, self.additional_files_to_register = dispatch_transfer.run(params)
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/bertha/compute/workflow/component/basecomponent.py", line 41, in run
    self.main(params)
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/bertha/compute/workflow/component/dispatch_transfer/component.py", line 323, in main
    self.get_relative_directory(params, self._get_illumina_date_string(current_date))
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/bertha/compute/workflow/component/dispatch_transfer/component.py", line 1056, in create_and_register_dispatch_data
    cip_api_client = self.get_cip_api_client()
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/bertha/compute/workflow/component/dispatch_transfer/component.py", line 919, in get_cip_api_client
    password=CIP_API_PASSWORD
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/pycipapi/cipapi_client.py", line 31, in __init__
    self.set_authenticated_header()
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/pycipapi/rest_client.py", line 102, in set_authenticated_header
    self.token = self.get_token()
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/pycipapi/cipapi_client.py", line 37, in get_token
    'password': self.password
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/pycipapi/rest_client.py", line 147, in post
    response = self._request_call('post', url, params=params, files=files, payload=payload)
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/pycipapi/rest_client.py", line 143, in _request_call
    return request_method(url, json=payload, params=parameters, headers=self.headers)
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/requests/sessions.py", line 518, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/requests/sessions.py", line 585, in send
    r = adapter.send(request, **kwargs)
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/requests/adapters.py", line 403, in send
    timeout=timeout
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py", line 623, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/tools/bertha-test/bertha-compute/1.28.0rc4/local/lib/python2.7/site-packages/requests/packages/urllib3/util/retry.py", line 236, in increment
    total -= 1
TypeError: unsupported operand type(s) for -=: 'Retry' and 'int'
``` 
using `requests==2.10.0` and `urllib3==1.22`. These are quite old versions but there's no minimum version of requests specified within `setup.py`. If you want me to drill down and figure out why this is necessary then I can do

Reply:
Thanks @Staberinde , let me check at the versions